src/commands/preview/dev_server.rs

```diff
@@ -0,0 +1,190 @@
+use std::net::{SocketAddr, ToSocketAddrs};
```

Comment:
because it is so specialized, and could turn into a weird part of the codebase for onboarding, perhaps this module deserves some heavy use of comments.